node/pod.go

```diff
@@ -374,17 +378,25 @@ func (pc *PodController) deletePodHandler(ctx context.Context, key string) (retE
 		span.SetStatus(err)
 		return err
 	}
-
+	if string(k8sPod.UID) != uid {
```

Comment:
Why not just rely on the precondition check?

Reply:
If we can not pass this check, we do not need send a request to k8s again.

Reply:
The request is not send to API server directly but instead to the local cache which IIRC will validate the `deleteOptions`.

Reply:
@pires Are there any docs about that? Can't that lead to a different kind of race?

Reply:
I just realized we're calling the `client` directly here, so it's a REST call and no cache is involved.

Reply:
I think it's fine to send a request to Kubernetes API as this is a corner case and after the key is processed it shouldn't show up again.